src/middlewared/middlewared/alembic/versions/12.1/2021-01-20_10-19_autoincrement.py

```diff
@@ -0,0 +1,65 @@
+"""Make all primary keys autoincrement
+
+Revision ID: 2fb0f87b2f17
+Revises: c68c71c34771
+Create Date: 2021-01-20 10:19:30.500426+00:00
+
+"""
+import re
+
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '2fb0f87b2f17'
+down_revision = 'c68c71c34771'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    conn = op.get_bind()
+    conn.execute("PRAGMA legacy_alter_table = TRUE")
+    for name, sql in conn.execute("SELECT name, sql FROM sqlite_master WHERE type = 'table'").fetchall():
+        if m := re.match(r'CREATE TABLE "(.+)" \(\s*"?id"? integer (NOT NULL |)PRIMARY KEY,', sql, flags=re.IGNORECASE):
```

Comment:
Wonder if there is any benefit in changing all these regex to be non-greedy (`.+?`)

Reply:
This is ran against a fixed set of strings every time and it already works, so I think there is not :)